lib/dao.js

```diff
@@ -1814,12 +1846,28 @@ DataAccessObject._coerce = function(where, options) {
         } else if (allowExtendedOperators && typeof val === 'object') {
           // Do not coerce object values when extended operators are allowed
         } else {
+          if (!allowExtendedOperators) {
+            var extendedOperators = Object.keys(val).filter(function(k) {
+              return k.indexOf('$') === 0;
+            });
+            if (extendedOperators.length) {
```

Comment:
Have you considered using [Array.prototype.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) to save unnecessary allocation of `extendedOperators` array?

Also the check for `$` at position 0 can be simplified as `k[0] === '$'`.

```js
const hasExtendedOperators = Object.keys(val).some(function(k) {
  return k[0] === '$';
});

if (hasExtendedOperators) {
// ...
```

Reply:
I wanted to report all violated operators instead of just a flag

Reply:
Makes sense 👍